python/tank/util/local_file_storage.py

```diff
@@ -84,7 +84,9 @@ def get_global_root(cls, path_type, generation=CORE_V18):
                            which is the current generation of paths.
         :return: Path as string
         """
-        if generation == cls.CORE_V18:
+        # If SHOTGUN_HOME is set then the intent is to not use any of official locations and instead
+        # use an alternate location, in which case this is as if we're running core 0.18.
+        if generation == cls.CORE_V18 or os.environ.get("SHOTGUN_HOME"):
```

Comment:
What's the reason we are making this change? Is it because of the session cache? Maybe we can add it to the comments? I am not sure i understand the motivation for this change - will it break backwards compatibility? 

To answer your question, i don't think we should ideally not break backwards comp.